src/alt.js

```diff
@@ -34,6 +34,13 @@ function uid(container, name) {
   return key
 }
 
+/* for now just detect whether it's instance of Immutable.js */
+function isImmutable(state) {
```

Comment:
The check being in here isn't what bothers me, it's the check for toJS being a function.

The con with this is that anyone can now define toJS as a function as part of their state and this will affect how getState() and setState behave.